test/servinge2e/helpers.go

```diff
@@ -31,15 +38,38 @@ func WaitForRouteServingText(t *testing.T, caCtx *test.Context, routeURL *url.UR
 		pkgTest.EventuallyMatchesBody(expectedText),
 		"WaitForRouteToServeText",
 		true,
-		insecureSkipVerify(),
+		addRootCAtoTransport(context.Background(), t.Logf, caCtx.Clients),
 	); err != nil {
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeURL, expectedText, err)
 	}
 }
 
-func insecureSkipVerify() spoof.TransportOption {
+// addRootCAtoTransport returns TransportOption when HTTPS option is true. Otherwise it returns plain spoof.TransportOption.
+func addRootCAtoTransport(ctx context.Context, logf logging.FormatLogger, clients *test.Clients) spoof.TransportOption {
```

Comment:
Can we not use upstream's `AddRootCAToTransport` for this?

Reply:
Yeah, I wanted to use upstream's `AddRootCAtoTransport` but the `Clients` which is the third argument of the func causes the type mismatch.

We defined our own `Clients` https://github.com/openshift-knative/serverless-operator/blob/4cdbaafd13b3ad84346a2b3cd8f91cc2ff40d8d4/test/clients.go#L38 and I think it is not easy to change.

Reply:
I see, that makes sense. I think the following should work tho

```golang
servingTest.AddRootCAtoTransport(context.Background(), t.Logf, &servingTest.Clients{KubeClient: caCtx.Clients.Kube}, true),
```

Ultimately, the function only needs the KubeClient, so passing this specific object through should be sufficient, I think.

Reply:
Oh! Thank you :bow: It works :heart_eyes:  Updated.